python/ray/data/dataset.py

```diff
@@ -912,138 +919,7 @@ def aggregate(self, *aggs: AggregateFn) -> U:
         ret = self.groupby(None).aggregate(*aggs).take(1)
         return ret[0] if len(ret) > 0 else None
 
-    def _check_and_normalize_agg_on(self,
```

Comment:
Moved to bottom of file and simplified since error handling is done is validate_key_fn.